### PR TITLE
TCES 61 User, Employer Contact, Error Screen Routing, 

### PR DESCRIPTION
**Pull request by kevinle623**
# Description

Implemented Employer Contact and Error Screen Routing. I already implemented and merged the User routing a very long time ago in the first routing pr.

For the Employer Contact, you are now able to edit, view, and add new contacts on the Employer page and also while adding a new employer too.

For the Error Screen Routing, I have already been setting up the <ErrorScreen> component across all error handling cases as I was implementing the routing from way before so I just replaced it with the ui design Error Screen using the 500 error screen (only case). 

**IMPORTANT NOTE** this pr has a fix to the crashing when job leads are added on employer add and regular job lead add 

Closes TCES-61

## Gallery 

image

image

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist

Please delete options that are not applicable.

- [x] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] New and existing unit tests pass locally with my changes (required)

### CODE REVIEW

File: client/src/components/shared/error-screen-component/index.jsx [hunk 1]
```diff
@@ -1,22 +1,9 @@
-import PropTypes from "prop-types";
-import { useNavigate } from "react-router-dom";
-import { ErrorContainer, ErrorMessage, BackButton } from "./index.styles";
-
-function ErrorComponent({ message }) {
-  const navigate = useNavigate();
+import Error500Component from "../error-components/500-error-component";
```
Comment (jordanjanakievski) [inline client/src/components/shared/error-screen-component/index.jsx]: There is a created page that can be used for the 500 and other error pages

**Review by jordanjanakievski — COMMENTED** [review]
This is all looking good. One small comment, and just missing a couple of routes
- User Profile -> when clicking the settings button
- 404 Error (pages/errors/404-error) -> Need a catch all for routes that don't exist
- Cookie Error (pages/errors/cookie-error)

**Review by jordanjanakievski — DISMISSED** [review]
Everything looks good.
Thanks for getting those comments addressed so quickly

**Review by DanielDervishi — APPROVED** [review]
Great Work Kevin!

**Review by jordanjanakievski — APPROVED** [review]
Looks good

**Pull request merged**